react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js

```diff
@@ -28,7 +28,13 @@ function reducer(state = initialState, action) {
     case actions.REMOVE_ITEM:
       return {
         ...state,
-        bookSelected: state.bookSelected.filter(book => book.id !== action.payload)
+        bookSelected: state.bookSelected
```

Comment:
you are iterating twice the same array (one with map then with filter), its better to do everything you can in one single iteration, see if you can find a way to end with the results you want with another array primitive.. perhaps [`reduce`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) might do the trick 😉